apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx

```diff
@@ -43,188 +36,214 @@ export interface RepeatingCustomEvent {
 interface CustomEventDialogProps {
     customEvent?: RepeatingCustomEvent;
     onDialogClose?: () => void;
-    scheduleNames: string[];
-}
-
-interface CustomEventDialogState extends RepeatingCustomEvent {
-    open: boolean;
-    scheduleIndices: number[];
 }
 
-const defaultCustomEvent: RepeatingCustomEvent = {
+const defaultCustomEventValues: RepeatingCustomEvent = {
     start: '10:30',
     end: '15:30',
     title: '',
     days: [false, false, false, false, false, false, false],
     customEventID: 0,
 };
 
-class CustomEventDialog extends PureComponent<CustomEventDialogProps, CustomEventDialogState> {
-    state: CustomEventDialogState = {
-        open: false,
-        ...(this.props.customEvent || defaultCustomEvent),
-        scheduleIndices: [],
-    };
+function CustomEventDialogs(props: CustomEventDialogProps) {
+    const [skeletonMode, setSkeletonMode] = useState(AppStore.getSkeletonMode());
+
+    const [open, setOpen] = useState(false);
+    const [scheduleIndices, setScheduleIndices] = useState<number[]>([]);
+    const [start, setStart] = useState(defaultCustomEventValues.start);
+    const [end, setEnd] = useState(defaultCustomEventValues.end);
+    const [title, setTitle] = useState(defaultCustomEventValues.title);
+    const [days, setDays] = useState(defaultCustomEventValues.days);
 
-    handleOpen = () => {
-        this.setState({ open: true, scheduleIndices: [AppStore.schedule.getCurrentScheduleIndex()] });
+    useEffect(() => {
+        const handleSkeletonModeChange = () => {
+            setSkeletonMode(AppStore.getSkeletonMode());
+        };
+
+        AppStore.on('skeletonModeChange', handleSkeletonModeChange);
+
+        return () => {
+            AppStore.off('skeletonModeChange', handleSkeletonModeChange);
+        };
+    }, []);
+
+    const handleOpen = useCallback(() => {
+        setOpen(true);
+        setScheduleIndices([AppStore.schedule.getCurrentScheduleIndex()]);
 
         logAnalytics({
             category: analyticsEnum.calendar.title,
             action: analyticsEnum.calendar.actions.CLICK_CUSTOM_EVENT,
         });
-    };
+    }, [setOpen]);
```

Comment:
You can remove this since `setOpen` should never change, and similar idea with other dependency arrays where applicable.

```suggestion
    }, []);
```